flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/cursor/MessageIdUtils.java

```diff
@@ -55,17 +59,15 @@ public static MessageId nextMessageId(MessageId messageId) {
      * message id. We don't support the batch message for its low performance now.
      */
     public static MessageIdImpl unwrapMessageId(MessageId messageId) {
-        MessageIdImpl idImpl = MessageIdImpl.convertToMessageIdImpl(messageId);
+        MessageIdImpl idImpl = convertToMessageIdImpl(messageId);
         if (idImpl instanceof BatchMessageIdImpl) {
             int batchSize = ((BatchMessageIdImpl) idImpl).getBatchSize();
-            checkArgument(batchSize == 1, "We only support normal message id currently.");
+            checkArgument(
+                    batchSize <= 1,
```

Comment:
OK. I get the point. So here we should use the condition: `batchSize == 1 || batchSize == 0`?

Reply:
Yep.

Reply:
We can't promise the batch size must be 0 or 1. Since this should be a mistake from Pulsar side that they return a `BatchMessageId` instead of `MessageId` for normal consuming. We should assume that batch size should below or equal to 1.